python/triton/runtime/cache.py

```diff
@@ -82,14 +84,14 @@ def get_group(self, filename: str) -> Optional[Dict[str, str]]:
     # Note a group of pushed files as being part of a group
     def put_group(self, filename: str, group: Dict[str, str]):
         if not self.cache_dir:
-            return
+            raise RuntimeError("Could not create or locate cache dir")
```

Comment:
I also make this more defensive. Let me know if this is ok (seems fine as long as all tests pass - it will cause error if you return None anyway).